src/formatter.jl

```diff
@@ -1,39 +1,61 @@
 using JuliaFormatter
 
+@static if isdefined(JuliaFormatter, :CONFIG_FILE_NAME) &&
+           isdefined(JuliaFormatter, :kwargs) &&
+           isdefined(JuliaFormatter, :parse_config) &&
+           isdefined(JuliaFormatter, :overwrite_options)
+  using JuliaFormatter: CONFIG_FILE_NAME, kwargs, parse_config, overwrite_options
+else
+  const CONFIG_FILE_NAME = ".JuliaFormatter.toml"
+
+  function kwargs(dict)
+      ns = (Symbol.(keys(dict))...,)
+      vs = (collect(values(dict))...,)
+      return pairs(NamedTuple{ns}(vs))
+  end
+
+  function parse_config(tomlfile)
+    config_dict = Pkg.TOML.parsefile(tomlfile)
+    if (style = get(config_dict, "style", nothing)) !== nothing
+      @assert (style == "default" || style == "yas") "currently $(CONFIG_FILE_NAME) accepts only \"default\" or \"yas\" for the style configuration"
+      if style == "yas" && isdefined(JuliaFormatter, :YASStyle)
+        config_dict["style"] = JuliaFormatter.YASStyle()
+      end
+    end
+    return kwargs(config_dict)
+  end
+
+  overwrite_options(options, config) = kwargs(merge(options, config))
+end
+
```

Comment:
If we're okay to just specify `JuliaFormatter = "^0.4.3"`, we don't need this backward compat check

Reply:
I don't think there is a motivation to use JuliaFormatter<4.3, so going to give up this.